### PR TITLE
feat(_layouts/page): add page layout with optional table of contents

### PR DESCRIPTION
**Pull request by feltnerm**
jekyll-toc is required to use the ToC

### CODE REVIEW

File: _sass/_toc.scss [hunk 1]
```diff
@@ -0,0 +1,12 @@
+.toc {
+  background-color: #f6f7f9;
+  padding: 8px;
+}
+
+.toc > ul,li{
```
Comment (mskelton) [inline _sass/_toc.scss]: I don't think you need to set list-style-type on the `li` element.

```suggestion
.toc > ul {
```

**Review by mskelton — DISMISSED** [review]

**Review by twilco — DISMISSED** [review]

**Comment by dbeg** [comment]
Could/should the `jekyll-toc` plugin be included in this theme? Any compelling reason to not include it?

**Comment by feltnerm** [comment]
@dbeg 

> Could/should the jekyll-toc plugin be included in this theme? Any compelling reason to not include it?

I'm down to include it. I was following the seemingly pre-existing conventions of the theme consumers adding the plugins. Not sure what the best way to do it is.

File: _sass/_toc.scss (hunk 1 above)
Comment (feltnerm) [reply _sass/_toc.scss]: Yeah I think my selector was bonkers. 1e3c01e seemed to do the trick.

File: _sass/_toc.scss [hunk 2]
```diff
@@ -0,0 +1,14 @@
+.toc {
+  background-color: #f6f7f9;
+  padding: 8px;
+}
+
+.toc {
+    ul, li {
```
Comment (jweaver-widen) [inline _sass/_toc.scss]: could maybe move the ul, ui targeted styles into the above block which is also .toc ?

Comment (feltnerm) [reply _sass/_toc.scss]: 💥 25df2b8

Comment (jweaver-widen) [reply _sass/_toc.scss]: woot woot!! 😀

**Review by jweaver-widen — DISMISSED** [review]

**Review by dbeg — APPROVED** [review]

**Pull request merged**